scheduler/scheduler-server/src/test/java/functionaltests/job/TestExecuteScriptsOnBusyNode.java

```diff
@@ -0,0 +1,301 @@
+import org.ow2.proactive.scripting.SelectionScript;
+
+import functionaltests.utils.SchedulerFunctionalTestWithCustomConfigAndRestart;
+import functionaltests.utils.SchedulerTHelper;
+import functionaltests.utils.TestNode;
+
+
+/**
+ * Test checks that when executing "node scripts" on busy nodes, the script execution does not impact the task occupying it
+ *
+ * It checks as well that a task starting on a node currently executing a "node script" does not impact its execution
+ * 
+ * @author ProActive team
+ *
+ */
+public class TestExecuteScriptsOnBusyNode extends SchedulerFunctionalTestWithCustomConfigAndRestart {
+
+    private static final long TIMEOUT = 60000;
+
+    private static int startedNodesCounter;
+
+    ExecutorService executorService;
+
+    public static class TestJavaTaskWithLock extends JavaExecutable {
+
+        private String fileLockPath;
+
+        @Override
+        public Serializable execute(TaskResult... results) throws Throwable {
+            getOut().println("OK");
+            FileLock.waitUntilUnlocked(fileLockPath);
+            return "OK";
+        }
+
+    }
+
+    public static class TestJavaTask extends JavaExecutable {
+
+        private String fileLockPath;
+
+        @Override
+        public Serializable execute(TaskResult... results) throws Throwable {
+            getOut().println("OK");
+            return "OK";
+        }
+
+    }
+
+    /**
+     * Starts an scheduler in non-fork mode with an empty rm
+     *
+     * @throws Exception
+     */
+    @BeforeClass
+    public static void startDedicatedScheduler() throws Exception {
+        schedulerHelper = new SchedulerTHelper(false,
+                                               new File(SchedulerTHelper.class.getResource("/functionaltests/config/scheduler-nonforkedscripttasks.ini")
+                                                                              .toURI()).getAbsolutePath(),
+                                               null,
+                                               null);
+    }
+
+    @Before
+    public void deployNode() throws Exception {
+        startNode();
+        executorService = Executors.newSingleThreadExecutor();
+    }
+
+    @Test
+    public void testExecuteScriptOnBusyNode() throws Exception {
+        FileLock fileLock = new FileLock();
+        Path fileLockPath = fileLock.lock();
+
+        log("Submit job");
+        final JobId jobId = schedulerHelper.submitJob(createJobWithLock(fileLockPath.toString()));
+
+        log("Wait when node becomes busy");
+        RMNodeEvent event;
+        do {
+            event = schedulerHelper.waitForAnyNodeEvent(RMEventType.NODE_STATE_CHANGED, TIMEOUT);
+        } while (!event.getNodeState().equals(NodeState.BUSY));
+
+        log("Wait when task starts");
+        schedulerHelper.waitForEventTaskRunning(jobId, "Test task");
+
+        log("Wait some time until task launcher is started");
+        Thread.sleep(5000);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)